src/containers/CourseEditor.js

```diff
@@ -107,6 +90,32 @@ class CourseEditor extends Component {
     })
   }
 
+  // Function for VIDEOS
+
+  uploadVideo = (video) => {
+    this.props.uploadVideo(video).then(data => {
+      this.state.course.videos.push(data.value);
```

Comment:
@TheoPth @NicolasGuary You should NEVER edit the state like this!!
https://reactjs.org/docs/state-and-lifecycle.html#using-state-correctly
When changing the state with something else than `this.setState` React is not capable of rerendering the component.
I am working on this branch so don't change it, just FYI.